### DISCUSSION TITLE
Documentation enhancement

### DISCUSSION THREAD

**Discussion by andrecasal** [brophdawg11/remix-validity-state#52] [General]
Hi @brophdawg11! Just wanted to reach out and let you know that I think the documentation could use a little touch-up. From what I understand, the first section mentions the `<FormProvider>` and `<Input>` components, which are syntactic sugar for `useValidatedInput()`. I think most developers, like myself, would want more direct control over their form, label, and inputs, so they might skip straight to using `useValidatedInput()`.

Also, it might be helpful to include an example for using custom refs. And, just a small change, instead of having the documentation say that the form should go inside the `FormProvider`, it could say the inputs should go inside instead. Something like this:

```tsx
import { FormProvider } from "remix-validity-state";

function MyFormPage() {
  return (
    <FormProvider formDefinition={formDefinition}>
      {/* Your form inputs go in here */}
    </FormProvider>
  );
}
```
Just some suggestions to make the documentation even clearer for developers.

I'd be happy to create issues for each of these and send you pull requests. Let me know what you think!

**Comment by brophdawg11** [comment]
Good ideas - I'm definitely happy to accept/review PRs you think would make it clearer.  Thanks!

**Reply by andrecasal** [reply]
Apologies for delay, have been quite busy. I haven't forgotten this 😊
